### ISSUE TITLE
[Discussion?] about converting outputs to fp32

### ISSUE THREAD

**Issue by cyr0930** [huggingface/accelerate#3477]
I found this logic (https://github.com/huggingface/accelerate/blob/v1.6.0/src/accelerate/accelerator.py#L1528) while I check memory footprint of my training with transformer model.

Loss functions in transformer have its own fp32 casting (https://github.com/huggingface/transformers/blob/v4.50.3/src/transformers/loss/loss_utils.py#L43), and also their output not only can have logits but also attentions, past_key_values, hidden_states.

As a result, accelerate redundantly casting logits again.

My question is
1. Is it responsible for accelerate to do the output casting?
2. I think the model outputs doesn't always have to be float32. (As the case ForSequenceClassificationLoss and ForQuestionAnsweringLoss in transformers shows no logit casting)

Maybe there are some history I don't know. Any opinion about this would be great. Thanks.

**Comment by cyr0930** [comment]
hmm looks like it depends on who's in charge of the mixed precision setting during training as both acceleratir and hfTrainer can set mp.
I think I should think about this more deeply. Thanks!

**Comment by SunMarc** [comment]
Closing for now !
